internal/lsp/server.go

```diff
@@ -37,24 +38,31 @@ func SendNotification(method string, params interface{}) {
 
 func RunServer[State any](args ServerArgs[State]) {
 	buf := NewMessageBuffer(os.Stdin)
+	mu := sync.Mutex{}
 
 	for {
 		request := buf.Read()
```

Comment:
Are you sure the request object is thread sage?

Reply:
I cannot see why not
The `request` object is just a plain object which is only read (is the result of the parsing of a json-rpc input payload)
besides, it's accessed by only one goroutine anyway

Reply:
You have :
```
bytes, err := json.Marshal(args.Handler(request, &args.InitialState))
if err != nil {
	panic(err)
}
```
in the go routine, so the object is accessible by multiple go routine.
But if the object is only read, I guess it is ok.